website/docs/API.md

```diff
@@ -669,3 +680,23 @@ it('should use context value', () => {
   // ...
 });
 ```
+
+## Accessibility
+
+### `isInaccessible`
+
+```ts
+function isInaccessible(
+  element: ReactTestInstance | null
+): boolean {}
+```
+
+Checks if given element should be excluded from the accessiblity e.g. by screen readers. 
+
+Element is considered inaccessbile when one of the follwing applies:
+* element has `display: none` style
```

Comment:
Should we also exclude `accessible={false}`?

Reply:
As far as I understand the documentation around [`accessible`](https://reactnative.dev/docs/accessibility#accessible) prop, it is not about marking element as accessible or not, but rather specifying that given element is an "accessibility element": "When a view is an accessibility element, it groups its children into a single selectable component. By default, all touchable elements are accessible."

So setting it to `false` does not mean that element is not accessible, but rather that it's not an accessibility element.

Reply:
`accessible={false}` implies that a component will not be read by a screen-reader as a touchable if I'm . A common use case is nested touchables, where you don't want the top one to be read.

Reply:
Your comments make sense, seems like this might be more complex functionality then I thought. I will convert this PR back to draft and read more about accessibility. Thanks for pointing out holes in my implementation.

Reply:
@AugustinLF I've one my research on a11y and here are my findings:

### Accessibility tree:
1. Accessibility tree is a parallel component tree to host component tree that consists only elements with which assistive technology can interact with (aka accessible objects): https://www.w3.org/TR/wai-aria-1.2/#terms 
2. Accessibility tree does not contain options that are purely presentational (`presentational`/`none` `role`), as they are not helpful to assistive technologies. So our regular `Views` that are not `buttons`, etc., will fit here.
3. Elements can be excluded from accessibility tree for [two reasons](https://www.w3.org/TR/wai-aria-1.2/#tree_exclusion):
  a. they are not to be displayed (`display: none`, `hidden` prop, `aria-hidden` prop, etc) - this condition applies to all of their children down the element tree
  b. they are presentational (`role="presentational"|"none"` - btw these are synonyms)

### [`isInaccessible`](https://testing-library.com/docs/dom-testing-library/api-accessibility) API from DTL:
1. It returns `true` only for elements that are not to be displayed (`display: none`, etc) but not for presentational elements. 

> This function will compute if the given element should be excluded from the accessibility API by the browser. It implements every MUST criteria from the [Excluding Elements from the Accessibility Tree](https://www.w3.org/TR/wai-aria-1.2/#tree_exclusion) section in WAI-ARIA 1.2 with the exception of checking the role attribute.

The `role` part is exactly about about `presentational/none` role not being excluded. [The function implementation](https://github.com/testing-library/dom-testing-library/blob/a9a8cf26992ff0f6b4257b7300939f461d04440d/src/role-helpers.js#L15) confirms this.

### Conclusions
1. There are three types of elements with respect o accessiblity:
 a. accessible elements - elements with which assistive technology can interact with
 b. presentational elements - elements which server layout/presentation purpose but are ignored by assistive tech
 c. hidden elements - elements which are explicitly excluded from assitive tech

2. DTL `isInaccessible` returns `true` for accessible & presentational elements, but `false` for hidden elements
3. RN's `accessible` would correspond to controlling whether element is accessible or presentational, so should not have an effect on our `isInaccessible` API. Similarly setting `importantForAccessibility` to `no` would only affect assignment to accessible or presentational groups.

Reply:
Great write up, you're right! I was not familiar enough with what actually `isInaccessible` entails, and what it want to cover. It does make sense when you come back to the root reason, which was detecting "hidden" elements, and in the case, `accessible:false` should indeed return that.